File: src/main/java/com/developersam/web/model/ten/TenBoardData.java
package com.developersam.web.model.ten;

/**
 * A data class that is responsible for holding the simplified version of the
 * ten board, without tracking extra game status. The class is designed as
 * a transmission object.
 */
final class TenBoardData {
    
    /**
     * The 9 * 9 board representation.
     */
    final int[][] board;
    /**
     * Current big square legal positions. Same as that in {@code TenBoard}.
     */
    final int currentBigSquareLegalPosition;
    /**
     * Current player identity.
     */
    final int currentPlayerIdentity;
    
    /**
     * A default constructor for GSON.
     */
    private TenBoardData() {
        board = null;
        currentBigSquareLegalPosition = -1;
        currentPlayerIdentity = -1;
    }
    
}

File: src/main/kotlin/com/developersam/ten/TenBoardData.kt
package com.developersam.ten

/**
 * A class that is responsible for holding the simplified version of the TEN
 * board, without tracking extra game status. The class is designed as
 * a transmission object.
 * - [board] describes the 9x9 board.
 * - [currentBigSquareLegalPosition] is used to determine the current game
 * status.
 * - [currentPlayerIdentity] is used to determine the identity of AI.
 */
class TenBoardData(
        internal val board: Array<IntArray>,
        internal val currentBigSquareLegalPosition: Int,
        internal val currentPlayerIdentity: Int
)